smartsim/settings/translators/launchArgTranslator.py

```diff
@@ -0,0 +1,219 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+import typing as t
+
+from smartsim.log import get_logger                                                                                    
+
+logger = get_logger(__name__)
+
+from ..common import IntegerArgument, StringArgument  
+
+
+class LaunchArgTranslator(ABC):
+    """Abstract base class that defines all generic launcher
+    argument methods that are not supported.  It is the
+    responsibility of child classes for each launcher to translate
+    the input parameter to a properly formatted launcher argument.
+    """
+
+    @abstractmethod
+    def launcher_str(self) -> str:
+        """ Get the string representation of the launcher
+        """
+        pass
+    
+    @abstractmethod
+    def set_reserved_launch_args(self) -> set[str]:
+        """ Reserved launch keys per Launcher.
+        """
+        pass
```

Comment:
Per offline discussion: it was decided to remove these `def set_*` methods from the abstract base class